### PR TITLE
Added Dartpad Sample to Hero Widget closes #69477

### LINKED ISSUE #69477: Adopt a Widget - Hero
The [Hero](https://master-api.flutter.dev/flutter/widgets/Hero-class.html) widget needs:

- A code / DartPad sample

For details about how to participate, see [flutter.dev/adoptawidget](https://flutter.dev/adoptawidget)

### PR DESCRIPTION
**Pull request by cv692001**
## Description

*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this PR is changing it, and what motivated the change. If you're changing visual properties, consider including before/after screenshots (and runnable code snippets to reproduce them).*

## Related Issues

*Replace this paragraph with a list of issues related to this PR from our [issue database]. Indicate, which of these issues are resolved or fixed by this PR. There should be at least one issue listed here.*

## Tests

I added the following tests:

*Replace this with a list of the tests that you added as part of this PR. A change in behavior with no test covering it
will likely get reverted accidentally sooner or later. PRs must include tests for all changed/updated/fixed behaviors. See [Test Coverage].*

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

closes #69477

### CODE REVIEW

File: packages/flutter/lib/src/widgets/heroes.dart [hunk 1]
```diff
@@ -77,8 +77,8 @@ Rect _boundingBoxFor(BuildContext context, [BuildContext? ancestorContext]) {
   final RenderBox box = context.findRenderObject()! as RenderBox;
   assert(box != null && box.hasSize);
   return MatrixUtils.transformRect(
-      box.getTransformTo(ancestorContext?.findRenderObject()),
-      Offset.zero & box.size,
+    box.getTransformTo(ancestorContext?.findRenderObject()),
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/heroes.dart:80]: can you please revert all the formatting changes in this PR?

**Comment by goderbauer** [comment]
Once the formatting changes are reverted, the analyzer check should also be happy with this.

**Comment by cv692001** [comment]
> Once the formatting changes are reverted, the analyzer check should also be happy with this.

Can you please help me how to revert the formatting changes , actually I am new to contributing :)

File: packages/flutter/lib/src/widgets/heroes.dart [hunk 2]
```diff
@@ -111,6 +111,116 @@ Rect _boundingBoxFor(BuildContext context, [BuildContext? ancestorContext]) {
...
+///           ),
+///         ],
+///       ),
+///     );
+///   }
+///
+///   void _gotoDetailsPage(BuildContext context) {
+///    Navigator.of(context).push(MaterialPageRoute(
+///      builder: (ctx) => Scaffold(
+///        body: Center(
```
Comment (johnpryan) [inline packages/flutter/lib/src/widgets/heroes.dart:204]: I would add an AppBar so that the user can navigate back

**Review by johnpryan — CHANGES_REQUESTED** [review]
Looks good overall, it would be great if this were a DartPad sample.

**Comment by johnpryan** [comment]
@cv692001 the easiest way is to probably re-apply your changes and make sure you're not running the formatter.

**Comment by goderbauer** [comment]
@@cv692001 Do you still have plans to follow up on the feedback?

**Comment by yash1200** [comment]
@goderbauer If the assignee doesn't respond, can you assign me for the issue?

**Comment by goderbauer** [comment]
I am going to close this one since there hasn't been any activity. Feel free to re-open when you have time to work on this again.

**Pull request closed without merging**